Asterix/WSC_functions.py

```diff
@@ -143,67 +253,273 @@ def creatingInterractionmatrix(input_wavefront, testbed, dimEstim,
             pos_in_matrix += DM.basis_size
 
         else:
-            # Creating Interaction Matrix for thd DM if does not exist
-            init_pos_in_matrix = pos_in_matrix
-            print("")
-            print(fileDirectMatrix + " does not exists:")
+            # Finally we can measure the matrix if we
+
+            #measure the initial FP. This is normalised with the testbed.
+            G0 = proc.resampling(
+                testbed.todetector(
+                    entrance_EF=input_wavefront,
+                    voltage_vector=initial_DM_voltage,
+                    save_all_planes_to_fits=save_all_planes_to_fits,
+                    dir_save_all_planes=dir_save_all_planes), dimEstim)
+
+            if (initial_DM_voltage == 0.).all():
+                print("")
+                print("The matrix " + fileDirectMatrix + " does not exists")
+
             print("Start " + DM_name)
 
+            # we measure the phases of the Basis we will apply on the DM
+            if DM.basis_type == 'fourier':
+                sqrtnbract = int(np.sqrt(DM.total_act))
```

Comment:
déjà fait ailleurs cette sqrt ? Besoin de le faire 2 fois ?
Comme précédemment sqrt => we assume a square DM

Reply:
En fait je m'en sers uniquement pour generer, mesurer et sauver la base de Fourier. Dans cette partie je ne considère pas les DMs ronds, je fais comme s'il était carré (34x34) puis je retire les actus inactif ensuite